File: LV1/1.3/icpcawards/icpcawards.cpp
#include <cstdio>
#include <map>
#include <string>
using namespace std;

int main() {
    int n, cnt = 0;
    char u[100], v[100];
    map<string, bool> uv;
    scanf("%d", &n);

    for (int i = 0; i < n && cnt < 12; i++) {
        scanf("%s %s", u, v);
        if (!uv[u]) {
            uv[u] = true;
            printf("%s %s\n", u, v);
            cnt++;
        }
    }
}

File: LV1/1.3/icpcawards/icpcawards.go
package main

import "fmt"

func main() {
	var n int
	cnt := 0
	uv := make(map[string]bool)

	fmt.Scanf("%d", &n)

	for i := 0; i < n && cnt < 12; i++ {
		var u, v string
		fmt.Scanf("%s %s", &u, &v)

		if !uv[u] {
			uv[u] = true
			fmt.Println(u, v)
			cnt++
		}
	}
}
